### PR TITLE
Supporting environment variables through Cloud Build

### PR DESCRIPTION
**Pull request by gislawill**
Also cleaning up app engine configurations

### CODE REVIEW

File: Dockerfile_dev
Comment (gislawill) [inline Dockerfile_dev:1]: Doesn't seem worthwhile to maintain this but let me know if you see a need

File: app.yaml [hunk 1]
```diff
@@ -1,8 +0,0 @@
-runtime: custom
```
Comment (gislawill) [inline app.yaml:1]: This is specific to app engine which we're no longer using

File: build_and_deploy.sh
Comment (gislawill) [inline build_and_deploy.sh:1]: Please correct me if I'm mistaken but I don't believe this is necessary given the configurations in cloud console

**Comment by lowriech** [comment]
Okay give this a quick think.  Basically there is one pattern I want to support, which is to be able to arbitrarily deploy a new service hosting this application, using a $ENV parameter.  See `build_and_deploy.sh`.  Beyond that, `cloudbuild.yaml` is a good wrapper of that script since it can provide $BRANCH_NAME and other useful CI/CD parameters. 

The substitutions can happen one of two places, either in cloudbuild.yaml (in which case we can embed them in the build trigger), or in the deployment shell script.  But at it's base, it should be easy to kickoff a new deployment with the shell script.  

Also, I brought back `dev.Dockerfile`.  It's not immediately useful, but is nice to be able to contain development and becomes more useful if/when we start composing multiple services.  

Also also, can you remind me how `_SITE_GATING_MATCH` works?  I was able to get the gating page up, but couldn't get the password to match.

**Pull request merged**